pkg/clustermanager/manager.go

```diff
@@ -190,21 +190,22 @@ func (m *Manager) toRESTConfig(cluster *v3.Cluster) (*rest.Config, error) {
 		if err != nil {
 			return nil, err
 		}
+	} else {
+		tlsDialer, err = normalTLSDialer(clusterDialer, u.Hostname(), caBytes)
+		if err != nil {
+			return nil, err
+		}
 	}
 
 	rc := &rest.Config{
 		Host:        u.String(),
 		BearerToken: cluster.Status.ServiceAccountToken,
-		TLSClientConfig: rest.TLSClientConfig{
-			ServerName: u.Hostname(),
-			CAData:     caBytes,
-		},
-		Timeout: 30 * time.Second,
+		Timeout:     30 * time.Second,
+		Transport:   new(http.Transport),
```

Comment:
Creating a new transport here concerns me.  Why are you doing this and are there some defaults you should be setting